app.js

```diff
@@ -1,8 +1,82 @@
+/* eslint-disable no-unused-vars */
+
 // import functions and grab DOM elements
+import { renderIngredientLI, renderMealLI } from './utils.js';
+//grabbing form, ingredients, and meals(recipe)
+const form = document.getElementById('add-ingredient'); // grab form
+const ingredientsList = document.getElementById('ingredient-list'); // list of ingredients
+const mealList = document.getElementById('meal-list'); // grabbing meal list
+// remove save and grabbing meal
+const remove = document.getElementById('remove');
+
+const meal = document.getElementById('meal-name');
+const saveMeal = document.getElementById('save-meal');
+
+//let states the ingredients combined = meals and added to total meal 
+let ingredients = [];
+let meals = [];
+
+// set event listeners
+// IMPURE RENDER FUNCTIONS
+// YOUR CODE MUST CALL THESE FUNCTIONS
+form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    const data = new FormData(form);
+    const item = {
+        ingredients: data.get('ingredient'),
+        qty: data.get('qty'),
+        measurement: data.get('measurement'),
+    };
+    ingredients.push(item);
+    renderIngredients();
+    form.reset();
+});
```

Comment:
This is super close! The `add` function is not quite working due to your `add` ingredient button having an id of `remove` but not a separate remove button. Fixing that should resolve this issue.